python/paddle/tensor/math.py

```diff
@@ -5107,3 +5107,188 @@ def frexp(x, name=None):
 
     mantissa = paddle.where((x < 0), mantissa * -1, mantissa)
     return mantissa, exponent
+
+
+def _trapezoid(y, x=None, dx=None, axis=-1, sum_mode='sum'):
+    """
+    Integrate along the given axis using the composite trapezoidal rule.
+
+    Args:
+        y (Tensor): Input array to integrate. It's data type should be float32, float64.
+        x (Tensor, optional): The sample points corresponding to the :attr:`y` values.
+            It's data type should be float32, float64.If :attr:`x` is None,
+            the sample points are assumed to be evenly spaced :attr:`dx` apart. The
+            default is None.
+        dx (float, optional): The spacing between sample points when x is None. If neither x nor dx is provided then the default is dx = 1
+        axis (int, optional): The axis along which to integrate. The default is -1.
+        sum_mode (str): use a different summation.
+
+    Returns:
+        Tensor, Definite integral of :attr:`y` = n-dimensional array as approximated along a single axis by the trapezoidal rule.
+    """
+    if sum_mode == 'sum':
+        sum = paddle.sum
+    elif sum_mode == 'cumsum':
+        sum = paddle.cumsum
+
+    if not (x is None or dx is None):
+        raise ValueError("Not permitted to specify both x and dx input args.")
+    if y.dtype not in [paddle.float32, paddle.float64]:
```

Comment:
input `y` shall support paddle.float16, because `paddle.gather`, `paddle.diff`, `paddle.sum` and `paddle.cumsum` are all support paddle.float16